cmake/AdaptSimSYCL.cmake

```
@@ -0,0 +1,45 @@
+add_library(SYCL::SYCL INTERFACE IMPORTED GLOBAL)
+target_link_libraries(SYCL::SYCL INTERFACE SimSYCL::simsycl)
+# add_sycl_executable_implementation function
+# Builds a SYCL program, compiling multiple SYCL test case source files into a
+# test executable, invoking a single-source/device compiler
+# Parameters are:
+#   - NAME             Name of the test executable
+#   - OBJECT_LIBRARY   Name of the object library of all the compiled test cases
+#   - TESTS            List of SYCL test case source files to be built into the
+# test executable
+function(add_sycl_executable_implementation)
+    cmake_parse_arguments(args "" "NAME;OBJECT_LIBRARY" "TESTS" ${ARGN})
+    set(exe_name            ${args_NAME})
+    set(object_lib_name     ${args_OBJECT_LIBRARY})
+    set(test_cases_list     ${args_TESTS})
+
+    add_library(${object_lib_name} OBJECT ${test_cases_list})
+    add_executable(${exe_name} $<TARGET_OBJECTS:${object_lib_name}>)
+
+    # hipSYCL needs the macro to be called on both the object library (to
```

```suggestion
    # SimSYCL needs the macro to be called on both the object library (to
```
Typo - although I think this isn't even true?

---

Removed this, it is indeed not needed.